saml.Test/saml.Test/TestResponseClassSaml.cs

```diff
@@ -0,0 +1,120 @@
+
+            Assert.AreEqual("12345678-testing1-testing2", groups[0]);
+            Assert.AreEqual("87654321-testing4-testing3", groups[1]);
+        }
+
+        /// <summary>This test method loads an empty XML SAML response and a legitimate certificate
+        /// into the 'Response' class of the SAML library. The test ensures that the appropriate 
+        /// exception is thrown.</summary>
+        [TestMethod()]
+        public void testEmptyXml()
+        {
+            Response response = null;
+
+            response = new Response(Constants.VALID_CERTIFICATE);
+
+            string xml_contents = getResourceContents(Constants.EMPTY_XML_RESPONSE_RESOURCE);
+
+            var load_xml_exception = Assert.ThrowsException<XmlException>(() => { response.LoadXml(xml_contents); });
+            Assert.IsTrue(load_xml_exception is XmlException);
+            Assert.IsTrue(load_xml_exception.Message.StartsWith("Unexpected end tag"));
+        }
+
+        /// <summary>This test method loads an invalid certificate
+        /// into the 'Response' class of the SAML library. This method tests to see 
+        /// if the correct exceptions are thrown.</summary>
+        [TestMethod]
+        public void testInvalidCertificate()
+        {
+            Response response = null;
+
+            var exception = Assert.ThrowsException<LoadCertificateException>(() => { response = new Response(Constants.INVALID_CERTIFICATE);  });
+
+            Assert.IsTrue(exception is LoadCertificateException);
+        }
+
+        /// <summary>This test method tests whether an exception is raised when the XML returned by the IDP
+        /// is invalid i.e contains bad chars. 
+        /// For this method, a valid certificate is used.</summary>
+        [TestMethod]
+        public void testInvalidXml()
+        { 
+            Response response = null;
+
+            response = new Response(Constants.VALID_CERTIFICATE);
+
+            string xml_contents = getResourceContents(Constants.INVALID_XML_RESPONSE_RESOURCE);
+
+            var load_xml_exception = Assert.ThrowsException<XmlException>(() => { response.LoadXml(xml_contents); });
+
+            Assert.IsFalse(response.IsValid());
+            Assert.IsTrue(load_xml_exception is XmlException);
+        }
+
+        /// <summary>
+        /// This test ensures that the library generates the appropriate exception 
+        /// when an empty certificate is loaded.
+        /// </summary>
+        [TestMethod]
+        public void testResponseEmptyCertificate()
+        {
+            Response response = null;
+
+            var exception = Assert.ThrowsException<LoadCertificateException>(() => { response = new Response(Constants.EMPTY_CERTIFICATE);  });
+            Assert.IsTrue(exception is LoadCertificateException);
+        }
+
+        /// <summary> This function returns the contents of a resource file.</summary>
+        /// <param name="resource_path"></param>
+        /// <returns></returns>
```

Comment:
We prefer not to specify private, just type 

`string getResourceContents(string resource_path)`

and it will be private by default